### DISCUSSION TITLE
Questions on build documentation on Mastodon

### DISCUSSION THREAD

**Discussion by chunhuizhao478** [idaholab/moose#26722] [Q&A Getting Started] [answered]
Hi all, I have a maybe simple question about building documentation on Mastodon, I fork the Mastodon on my local app, with moose and blackbear placed aside, I'm able to build the executable. Then when I run "./moosedocs.py" returns error: modules are not found for "MooseDocs" and "mooseutils".

I try to change the python interpreter to use minforge/moose related ones, but it's not helping. I know the "MooseDocs" dir is under "moose/python", what should I do to link it to Mastodon? Thanks!

**Comment by cticenhour** [comment, marked as answer]
You should be able to add the python folder within moose to your `PYTHONPATH`. Much like your system `PATH`, it will check the directories placed there first for any modules. See the command you need to enter into your terminal below to set that before running moosedocs again (also make sure your moose environment is activated, as normal). Make sure you replace the sample path I use below with your actual path to `moose/python`!

```
export PYTHONPATH=/Users/username/projects/mastodon/moose/python
```

This will be unset whenever you close your Terminal, so you'll need to re-add it if you want to use it again.

**Reply by chunhuizhao478** [reply]
Thanks @cticenhour. I just find it a bit weird that after I git clone the mastodon repo, the "moose" and "blackbear" folders are empty, so I git clone both as well to place alongside with "mastodon" in the same folder. After I did when you suggest above, it pops up error about not finding "mastodon/moose/large_media", which makes sense because it's empty. How should I deal with it? Thanks again.

**Reply by GiudGiud** [reply]
Hello

Use the git submodule init and git submodule update commands to get these folders to be filled

**Reply by GiudGiud** [reply]
For large_media, it s a submodule of moose, which is itself a submodule of mastodon. You can recursively init and update submodules with `git submodule init moose --recursive` iirc

**Reply by chunhuizhao478** [reply]
Thanks @GiudGiud. it works fine now!
